pkg/member/member_control.go

```diff
@@ -265,16 +272,55 @@ func (m *memberControl) UpdateMember(ctx context.Context, cli client.ClusterClos
 	return m.updateMemberPeerAddress(ctx, cli, membersInfo.Header.GetMemberId())
 }
 
-func (m *memberControl) doPromoteMember(ctx context.Context, member *etcdserverpb.Member, cli client.ClusterCloser) error {
-	memPromoteCtx, cancel := context.WithTimeout(ctx, brtypes.DefaultEtcdConnectionTimeout)
+// RemoveMember removes the member from the etcd cluster.
+func (m *memberControl) RemoveMember(ctx context.Context) error {
+	m.logger.Infof("Removing the %v member from cluster", m.podName)
+
+	cli, err := m.clientFactory.NewCluster()
+	if err != nil {
+		return fmt.Errorf("failed to build etcd cluster client : %v", err)
+	}
+	defer cli.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, brtypes.DefaultEtcdConnectionTimeout)
 	defer cancel()
-	_, err := cli.MemberPromote(memPromoteCtx, member.ID) //Member promote call will succeed only if member is in sync with leader, and will error out otherwise
-	if err == nil {                                       //Member successfully promoted
-		m.logger.Infof("Member %v with ID: %v has been promoted", member.GetName(), member.GetID())
-		return nil
-	} else if errors.Is(err, rpctypes.Error(rpctypes.ErrGRPCMemberNotLearner)) { //Member is not a learner
-		m.logger.Info("Member ", member.Name, " : ", member.ID, " already part of etcd cluster")
-		return nil
+
+	memberInfo, err := cli.MemberList(ctx)
+	if err != nil {
+		return fmt.Errorf("error listing members: %v", err)
 	}
-	return err
+
+	foundMember := findMember(memberInfo.Members, m.podName)
+	if foundMember == nil {
+		return ErrMissingMember
+	}
+
+	return m.removeMemberFromCluster(ctx, cli, foundMember.GetID())
```

Comment:
so that we can able to write unit tests, as you can see `removeMemberFromCluster` takes the etcd client as an input, so the logic is to mock this etcd client and passed the mock client to this func. 
you can take look at these [unit tests](https://github.com/gardener/etcd-backup-restore/pull/509/commits/fb96427033e0276da7095e228151c16e9052f98e#diff-c999015e85f8489977cc7e1e2ef722767361b9a3524da0543443dec10f777d11R310-R369) which I added.